vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -170,14 +172,17 @@
             selectedKeys[item.key] = item;
             if (userOriginated) {
               item.selected = true;
-              grid.$server.select(item.key);
+              if (isNotSwitchNewAndOldValue) {
```

Comment:
This logic doesn't work! This flag is set to `false` when deselection is disabled and the user tries to deselect. The following selection of another item would not get communicated to the server because of the global falsy flag. Add a test for this case.

Rather fix this by allowing `activeItem` to be set `null` programmatically (by `doDeselection`).

Reply:
as discussion in person, I pushed the new fix with adding more test that covers the case: a selection of another item would get communicated to the server.